### PR TITLE
Todo list Dashboard - Implement functionality w/ dummy data

### LINKED ISSUE #62: Todo list (Dashboard / Full Screen View) - Implement functionality w/ dummy data
**Is your feature request related to a problem? Please describe.**
The todo list has no ability to track the action or event and needs work to read data

**Describe the solution you'd like**
Implement a way that enables tracking activities and events on todo list. This should use dummy data and storage to ensure correct functionality.

**Describe alternatives you've considered**
Not implementing functionality. This would lead to an unusable widget that serves no purpose.

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by yzho976**
Closes #[62](https://github.com/UOA-SE701-Group3-2021/3Lancers/issues/62)

## Proposed Changes

-Functionality with dummy data
Basic
image

### CODE REVIEW

**Comment by ysway** [comment]
Could you solve the conflict

**Review by JChamberlain28 — CHANGES_REQUESTED** [review]
As said above, there are conflicts to resolve. Please rebase and resolve.

**Comment by Milk-Yan** [comment]
Could you please include images of your changes?

**Comment by milkyway404** [comment]
Could you please include images and/or descriptions of your changes for before/after?

File: telos-frontend/src/components/todo/DashboardTodo.jsx
Comment (RawiriHohepa) [inline telos-frontend/src/components/todo/DashboardTodo.jsx:213]: This does not follow the design defined [in the wiki](https://github.com/UOA-SE701-Group3-2021/3Lancers/wiki/To-Do-List#hi-fi-design-2). If you wish to change the design, you should open a new issue and wait for it to be approved. Changing the design is not within the scope of this issue.

File: telos-frontend/src/components/todo/DashboardTodo.jsx
Comment (RawiriHohepa) [inline telos-frontend/src/components/todo/DashboardTodo.jsx]: These functions to change the data should have associated unit tests to check that data has actually been changed.

File: telos-frontend/src/components/todo/DashboardTodo.jsx
Comment (RawiriHohepa) [inline telos-frontend/src/components/todo/DashboardTodo.jsx]: It's generally bad practice to disable eslint rules - what's your reasoning for doing this here?

**Review by RawiriHohepa — CHANGES_REQUESTED** [review]
Hi Evan, well done working on the speed with which you did this, and the amount of effort you put in!

There's a few issues with your pull request that need to be changed or discussed before approval.
As a further note to my comments on the files changed, there is a merge conflict that must be resolved.

**Review by lhuijser — CHANGES_REQUESTED** [review]
I appreciate all of the work you've put into this, however, all the members of the team have worked hard to come to an agreement on the design and proposed functionality of this, so to run against this expectation wouldn't be fair on the team as a whole. Could you please switch back to the original design?

**Comment by yzho976** [comment]
> Could you please include images and/or descriptions of your changes for before/after?

Fixed

File: telos-frontend/src/components/todo/DashboardTodo.jsx
Comment (milkyway404) [inline telos-frontend/src/components/todo/DashboardTodo.jsx]: why is it 98%? Could you please include a comment about the reason in the file?

File: telos-frontend/src/components/todo/DashboardTodo.jsx
Comment (milkyway404) [inline telos-frontend/src/components/todo/DashboardTodo.jsx:115]: if this only toggles one value, then would it be better to just include the index?

File: telos-frontend/src/components/todo/DashboardTodo.jsx
Comment (milkyway404) [inline telos-frontend/src/components/todo/DashboardTodo.jsx]: what is the difference between itemList and listitems? Why do we need two different variables?

File: telos-frontend/src/components/todo/DashboardTodo.jsx
Comment (milkyway404) [inline telos-frontend/src/components/todo/DashboardTodo.jsx]: This should be cancelEvent

File: telos-frontend/src/components/todo/DashboardTodo.jsx
Comment (milkyway404) [inline telos-frontend/src/components/todo/DashboardTodo.jsx]: again, camelCase?

File: telos-frontend/src/components/todo/DashboardTodo.jsx
Comment (milkyway404) [inline telos-frontend/src/components/todo/DashboardTodo.jsx]: same as above

File: telos-frontend/src/components/todo/DashboardTodo.jsx
Comment (milkyway404) [inline telos-frontend/src/components/todo/DashboardTodo.jsx]: I feel that this probably isn't the best way to name the variable. Also, this probably isn't a deep copy ?

**Review by milkyway404 — CHANGES_REQUESTED** [review]
Functionality looks really good, just some confusions about variable choices and naming.

**Review by lhuijser — CHANGES_REQUESTED** [review]
The changes look really good, thank you for doing this :) Only issue I can see is that the 'new to do' part does not quite match up with the hi fi design on figma, and similarly the menu that shows up when the three dots is clicked is not quite the same as the design either. If you could have a look at it and make changes to match those then that would be great!

File: telos-frontend/src/components/todo/DashboardTodo.jsx
Comment (milkyway404) [inline telos-frontend/src/components/todo/DashboardTodo.jsx]: Could you import like:
import { Button, Dialog, etc } from '@material-ui/core'?

**Review by milkyway404 — COMMENTED** [review]
Looks generally good, left one comment that could be fixed

**Review by ysway — APPROVED** [review]
Looks good for me now

File: telos-frontend/src/components/todo/DashboardTodo.jsx
Comment (JChamberlain28) [inline telos-frontend/src/components/todo/DashboardTodo.jsx:342]: are we deciding not to validate inputs? Is date impacted by this?

**Review by JChamberlain28 — COMMENTED** [review]
I cannot comment on specific functionality as I am unfamiliar with the functionality of this widget. Please see my comment. Also you have requested changes to resolve.

File: telos-frontend/src/components/todo/DashboardTodo.jsx
Comment (yzho976) [reply telos-frontend/src/components/todo/DashboardTodo.jsx:342]: there is a date picker there so not really ;)

Comment (JChamberlain28) [reply telos-frontend/src/components/todo/DashboardTodo.jsx:342]: Also a little more commenting in the code to explain what certain things are would be useful

File: telos-frontend/src/components/todo/DashboardTodo.jsx
Comment (lhuijser) [inline telos-frontend/src/components/todo/DashboardTodo.jsx:2]: This is mainly out of curiosity, but why is eslint being disabled here?

File: telos-frontend/src/components/todo/DashboardTodo.jsx
Comment (lhuijser) [inline telos-frontend/src/components/todo/DashboardTodo.jsx:109]: Functionality looks good, however if the naming could be made a bit more clear or some comments could be added to explain what different things do that would be useful, as I'm not sure what things like this particular function represent

**Review by lhuijser — CHANGES_REQUESTED** [review]
The changes look good to me, just a couple of comments :)

File: telos-frontend/src/components/todo/DashboardTodo.jsx
Comment (RawiriHohepa) [reply telos-frontend/src/components/todo/DashboardTodo.jsx:2]: +1, I asked in a previous review but looks like the line I commented on was removed

**Review by lhsunyc — APPROVED** [review]
Looks good

**Review by lhuijser — APPROVED** [review]
Changes look great, thank you! Still one unresolved comment about eslint being disabled but this is minor and doesn't impact the functionality, so can be resolved later.

**Review by RawiriHohepa — APPROVED** [review]

**Pull request merged**
